components/tikv_util/src/quota_limiter.rs

```diff
@@ -0,0 +1,220 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::time::Duration;
+
+use super::config::ReadableSize;
+use super::time::Limiter;
+use super::timer::GLOBAL_TIMER_HANDLE;
+
+use cpu_time::ThreadTime;
+use futures::compat::Future01CompatExt;
+
+// The cpu time is not a real statistics, only part of the processing logic is
+// taken into account, so it needs to be multiplied by a factor.
+// transfer milli cpu to micro cpu
+//
+// TODO: Don't adjusted based on experience.
+const CPU_TIME_FACTOR: f64 = 0.8;
+
+// To avoid long tail latency.
+const MAX_QUOTA_DELAY: Duration = Duration::from_secs(1);
+
+// Quota limiter allows users to obtain stable performance by increasing the
+// completion time of tasks through restrictions of different metrics.
+#[derive(Debug)]
+pub struct QuotaLimiter {
+    cputime_limiter: Limiter,
+    write_bandwidth_limiter: Limiter,
```

Comment:
why do we need a write/read limiter here, I think io rate limiter is enough. /cc @tabokie

Reply:
The frontend read bandwidth may be much larger than the real disk read because of the existence of block cache. I'm not sure if that's the cause of a separate read limit.

Reply:
I do think bandwidth is somewhat misleading. When using the quota limiter, we don't really care about how many physical IOs (both buffered and on disk) are involved. We are just (blindly) tuning these knobs for better results. It's better to think of them as request size rather than IO resources.

Reply:
> 

good point. The name of these two limiter may be renamed to write_request_size_limiter, read_response_size_limiter..

Reply:
> why do we need a write/read limiter here, I think io rate limiter is enough. /cc @tabokie

The IO rate limiter is more about eliminate low priority task's impact to high priority task, per my understand. Here the read/write limiter is more about control frontend traffic. 
This is just in case CPU is not the bottleneck (but IO is).